maint/bumplibs.sh

```diff
@@ -148,18 +147,20 @@ process_lib() {
     HEADERS_DIFF="$(diff <(echo "$HEADERS_LAST") <(echo "$HEADERS_HEAD"))"
     HEADERS_GONE="$(echo "$HEADERS_DIFF" | sed -n -e 's/^< //p')"
     HEADERS_ADDED="$(echo "$HEADERS_DIFF" | sed -n -e 's/^> //p')"
-    HEADERS_COMMON="$(echo -e "$HEADERS_LAST\n$HEADERS_HEAD" | sed -n -e 's/^ *\(.*\) *$/\1/p' | sort | uniq -d)"
 
     # Check whether there were any changes to headers or sources
     SOURCES="$(find_sources "$LIB" "$AMFILE")"
-    CHANGES=$(git diff -w $LAST_RELEASE..HEAD $HEADERS_COMMON $SOURCES | wc -l)
-    if [ $CHANGES -eq 0 ]; then
-        if [ -z "$HEADERS_GONE" ]; then
-          echo "No changes to $LIB interface"
-          prompt_to_continue
-          echo ""
-          return
-        fi
+    if [ -n "$HEADERS_GONE" ]; then
+        DEFAULT_CHANGE="i" # Removed public header is incompatible change
+    elif [ -n "$HEADERS_ADDED" ]; then
+        DEFAULT_CHANGE="c" # Additions are likely compatible
+    elif git diff --quiet -w $LAST_RELEASE..HEAD $HEADERS_HEAD $SOURCES ; then
+        echo "No changes to $LIB interface"
+        prompt_to_continue
+        echo ""
+        return
+    else
+        DEFAULT_CHANGE="f" # Sources changed, so it's at least a fix
     fi
 
```

Comment:
Making an inspired guess here is a big improvement.
Bad impact - as far as I got it so far - is that picking up after interrupting bumplibs.sh has become more complicated.
iirc up to know you could just go with [None] and press <enter> to skip over the already entered ones.
Now an <enter> would mean to revert to the initial guess.
If the version in Makefile.am differs from that in HEAD we might derive the guess from that and could possibly mark it so that the user knows he is editing something he has already touched.
This would bring back resume-ability and improve overview and thus reduce the risk to mess up.
Of course not sure if we need it for this iteration. Some thoughts about how to improve going from one rc to the next might pay off as well.

Reply:
Good point, getting the current version from HEAD would be better. 

Yep, doing a little improvement each release helps a lot. I'll add that to the to-do for next time ...